src/Notification/NotificationSyncer.php

```diff
@@ -115,7 +112,16 @@ public function sync(Blueprint\BlueprintInterface $blueprint, array $users)
         // receiving this notification for the first time (we know because they
         // didn't have a record in the database).
         if (count($newRecipients)) {
-            $this->sendNotifications($blueprint, $newRecipients);
+            $this->queue->push(new SendNotificationsJob($blueprint, $newRecipients));
+        }
+
+        if ($blueprint instanceof MailableInterface) {
+            /** @var User $recipient */
+            foreach ($newRecipients as $recipient) {
+                if ($recipient->shouldEmail($blueprint::getType())) {
+                    $this->queue->push(new SendEmailNotificationJob($blueprint, $recipient));
+                }
+            }
```

Comment:
This whole block (starting at the check for `MailableInterface`) seems unnecessary now that the `SendNotificationsJob` dispatches the job for sending emails (which is good). Assuming the other comment about the missing loop gets tackled...